service/migration_manager.cc

```diff
@@ -164,7 +164,19 @@ void migration_manager::init_messaging_service()
                 }
             }
         }
+
         if (cm_retval_supported) {
+            // If the schema we're returning was last modified in group 0 mode, we also need to return
+            // the persisted schema version so the pulling node uses it instead of calculating a schema digest.
+            //
+            // If it was modified in RECOVERY mode, we still need to return the mutation as it may contain a tombstone
+            // that will force the pulling node to revert to digest calculation instead of using a version that it
+            // could've persisted earlier.
+            //
+            // We do it only in `cm_retval_supported` branch -- there is no way for GROUP0_SCHEMA_VERSIONING feature
+            // to be enabled without all nodes understanding `canonical_mutation`s.
+            cm.emplace_back(co_await self._sys_ks.local().get_group0_schema_version());
+
```

Comment:
This was added in 2019, 7adc764b6eff4e5a839ad387836d31d6ee92b40c. We can make it mandatory and drop the other branch.